lib-multisrc/madara/src/eu/kanade/tachiyomi/multisrc/madara/Madara.kt

```diff
@@ -958,9 +1109,8 @@ abstract class Madara(
             .toByteArray()
     }
 
-    override fun setupPreferenceScreen(screen: PreferenceScreen) {
-        addRandomUAPreferenceToScreen(screen)
-    }
+    protected fun launchIO(block: () -> Unit) =
```

Comment:
IDE says:
> This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
```suggestion
    @OptIn(DelicateCoroutinesApi::class)
    protected fun launchIO(block: () -> Unit) =
```

Any reason why this shouldn't be added?

Reply:
that propogates the warning to every call site of the function

Reply:
why not use a CoroutineScope instead of a GlobalScope then?

Reply:
I was under the impression it gets cancelled once function scope ends. turns out that isn't how it works.